blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/scm/Scm.java

```diff
@@ -73,4 +79,34 @@
     @PUT
     @WebMethod(name = VALIDATE)
     public abstract @CheckForNull HttpResponse validateAndCreate(@JsonBody JSONObject request);
+
+    /**
+     * Save a file to this repository. Creates a new one if it doesn't exist.
+     *
+     * @return {@link ScmFile}
+     */
+    @PUT
+    @WebMethod(name="")
+    @TreeResponse
+    public ScmFile saveFile(StaplerRequest staplerRequest) throws IOException {
+        JSONObject body = JSONObject.fromObject(IOUtils.toString(staplerRequest.getReader()));
+        String cls = (String) body.get("$class");
+        if(cls == null){
+            throw new ServiceException.BadRequestExpception(new ErrorMessage(400, "Failed to save file")
+                    .add(new ErrorMessage.Error("$class", ErrorMessage.Error.ErrorCodes.MISSING.toString(),
+                            "$class is required")));
+        }
+        ScmSaveFileRequest request = staplerRequest.bindJSON(ScmSaveFileRequest.class, body);
```

Comment:
We really should have an automatic way of doing this, no? Thinking of other users implementing APIs, there are much easier ways available with automatic object bindings, don't they exist in stapler?